dos/src/main.rs

```diff
@@ -211,232 +271,290 @@ fn run_dos(
-#[clap(rename_all = "kebab-case")]
-struct TransactionParams {
-    #[clap(
-        long,
-        default_value = "2",
-        help = "Number of signatures in transaction"
-    )]
-    num_signatures: usize,
-
-    #[clap(long, help = "Generate a valid blockhash for transaction")]
+fn create_payers<T: 'static + BenchTpsClient + Send + Sync>(
     valid_blockhash: bool,
-
-    #[clap(long, help = "Generate valid signature(s) for transaction")]
-    valid_signatures: bool,
-
-    #[clap(long, help = "Generate unique transactions")]
-    unique_transactions: bool,
-
-    #[clap(
-        long = "payer",
-        help = "Payer's keypair file to fund transactions [Optional]"
-    )]
-    payer_filename: Option<String>,
+    size: usize,
+    client: &Option<Arc<T>>,
+) -> Vec<Option<Keypair>> {
+    // Assume that if we use valid blockhash, we also have a payer
+    if valid_blockhash {
+        // each payer is used to fund transaction
+        // transactions are built to be invalid so the the amount here is arbitrary
+        let funding_key = Keypair::new();
+        let funding_key = Arc::new(funding_key);
+        let res = generate_and_fund_keypairs(
+            client.as_ref().unwrap().clone(),
+            &funding_key,
+            size,
+            1_000_000,
+        )
+        .unwrap_or_else(|e| {
+            eprintln!("Error could not fund keys: {:?}", e);
+            exit(1);
+        });
+        res.into_iter().map(Some).collect()
+    } else {
+        std::iter::repeat_with(|| None).take(size).collect()
+    }
 }
 
-#[derive(ArgEnum, Clone, Eq, PartialEq)]
-enum Mode {
-    Gossip,
-    Tvu,
-    TvuForwards,
-    Tpu,
-    TpuForwards,
-    Repair,
-    ServeRepair,
-    Rpc,
+fn get_permutation_size(num_signatures: Option<usize>, num_instructions: Option<usize>) -> usize {
+    if let Some(num_signatures) = num_signatures {
+        num_signatures
+    } else if let Some(num_instructions) = num_instructions {
+        num_instructions
+    } else {
+        1 // for the case AccountCreation
+    }
 }
 
-#[derive(ArgEnum, Clone, Eq, PartialEq)]
-enum DataType {
-    RepairHighest,
-    RepairShred,
-    RepairOrphan,
-    Random,
-    GetAccountInfo,
-    GetProgramAccounts,
-    Transaction,
-}
+fn run_dos_transactions<T: 'static + BenchTpsClient + Send + Sync>(
+    target: SocketAddr,
+    iterations: usize,
+    client: Option<Arc<T>>,
+    transaction_params: TransactionParams,
+) {
+    let socket = UdpSocket::bind("0.0.0.0:0").unwrap();
 
-fn addr_parser(addr: &str) -> Result<SocketAddr, &'static str> {
-    match solana_net_utils::parse_host_port(addr) {
-        Ok(v) => Ok(v),
-        Err(_) => Err("failed to parse entrypoint address"),
+    let num_signatures = transaction_params.num_signatures;
+    let num_instructions = transaction_params.num_instructions;
+    let valid_signatures = transaction_params.valid_signatures;
+    let valid_blockhash = transaction_params.valid_blockhash;
```

Comment:
Do you feel strongly about these reassignments? I find them more distracting than helpful, but open to counterarguments.

Reply:
Not really, I just though it is shorter but maybe  to verbose, fixed